### PR TITLE
fix: maintain order of indexed probabilities in oaa and csoaa

### PR DESCRIPTION
**Pull request by bassmang**
resolves: #3935

### CODE REVIEW

File: vowpalwabbit/core/src/reductions/oaa.cc [hunk 1]
```diff
@@ -34,10 +34,10 @@ struct oaa
   uint64_t num_subsample = 0;           // for randomized subsampling, how many negatives to draw?
   uint32_t* subsample_order = nullptr;  // for randomized subsampling, in what order should we touch classes
   size_t subsample_id = 0;              // for randomized subsampling, where do we live in the list
-  int indexing = -1;                    // for 0 or 1 indexing
   VW::io::logger logger;
+  uint32_t& indexing;  // for 0 or 1 indexing
```
Comment (lalo) [inline vowpalwabbit/core/src/reductions/oaa.cc:38]: can you add a comment for indexing == 2 or put it as a constant somewhere with a clear name?

File: vowpalwabbit/core/include/vw/core/global_data.h [hunk 2]
```diff
@@ -329,6 +329,8 @@ struct workspace
   // hack to support cb model loading into ccb reduction
   bool is_ccb_input_model = false;
 
+  uint32_t indexing = 2;
```
Comment (lalo) [inline vowpalwabbit/core/include/vw/core/global_data.h:332]: maybe add the comment here too

File: vowpalwabbit/core/src/parse_args.cc [hunk 3]
```diff
@@ -682,7 +682,8 @@ void parse_feature_tweaks(options_i& options, VW::workspace& all, bool interacti
                .help("Don't remove interactions with duplicate combinations of namespaces. For ex. this is a "
                      "duplicate: '-q ab -q ba' and a lot more in '-q ::'."))
       .add(make_option("quadratic", quadratics).short_name("q").keep().help("Create and use quadratic features"))
-      .add(make_option("cubic", cubics).keep().help("Create and use cubic features"));
+      .add(make_option("cubic", cubics).keep().help("Create and use cubic features"))
+      .add(make_option("indexing", all.indexing).one_of({0, 1}).keep().help("Choose between 0 or 1-indexing"));
```
Comment (lalo) [inline vowpalwabbit/core/src/parse_args.cc:686]: verify that the logic of one_of doesn't clash with the default value of 2 of indexing

Comment (bassmang) [reply vowpalwabbit/core/src/parse_args.cc:686]: Yea it works fine if the `indexing` option isn't supplied. 2 is meant as a placeholder if nothing is specified and shouldn't be an option for the user.

**Review by lalo — APPROVED** [review]

**Pull request merged**